src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/VBDefineConstantsEncoding.cs

```diff
@@ -71,28 +83,55 @@ static IEnumerable<string> ReadEntries(string rawText)
                         escaped = false;
                     }
                 }
-                throw new FormatException("Expected valid name value pair.");
+                throw new FormatException("Expected valid name value pair for defining custom constants.");
             }
 
-            static string DecodeCharacters(string value)
+            static string SplitSingleEntry(string entry)
             {
-
-                return value.Replace("/\"", "\"").Replace("/=", "=").Replace("/,", ",").Replace("//", "/");
+                bool escaped = false;
+                for (int i = 0; i < entry.Length; i++)
+                {
+                    if (entry[i] == '=' && !escaped)
+                    {
+                        throw new FormatException($"Expected valid name value pair for defining custom constants.");
+                    }
+                    else if (i == entry.Length - 1)
+                    {
+                        return entry;
+                    }
+                    else if (entry[i] == '/')
+                    {
+                        escaped = !escaped;
+                    }
+                    else
+                    {
+                        escaped = false;
+                    }
+                }
+                throw new FormatException("Expected valid name value pair for defining custom constants.");
             }
 
-            static string StripQuotes(string value)
+            static string DecodeCharacters(string value)
             {
-                if (value.StartsWith("\"") && value.EndsWith("\"") && value.Length >= 2)
-                {
-                    return value.Substring(1, value.Length - 2);
-                }
-                return value;
+                return value.Replace("/\"", "\"").Replace("/=", "=").Replace("/,", ",").Replace("//", "/");
             }
         }
 
         public string Format(IEnumerable<(string Name, string Value)> pairs)
         {
-            return string.Join(",", pairs.Select(pair => $"{EncodeCharacters(pair.Name)}=\"{EncodeCharacters(pair.Value)}\""));
+            List<string> values = new List<string>();
+            foreach ((string Name, string Value) in pairs)
+            {
+                if(string.IsNullOrEmpty(Value))
+                {
+                    values.Add(EncodeCharacters(Name));
+                }
+                else
+                {
+                    values.Add($"{EncodeCharacters(Name)}={EncodeCharacters(Value)}");
+                }
+            }
+            return string.Join(",", values);
```

Comment:
```suggestion
            return string.Join(",", FormatInternal(pairs));

            static IEnumerable<string> FormatInternal(IEnumerable<(string, string)> pairs)
            {
                foreach ((string name, string value) in pairs)
                {
                    string formattedValue = string.IsNullOrEmpty(value) ? $"={EncodeCharacters(value)}" : string.Empty;
                    yield return $"{EncodeCharacters(name)}{formattedValue}";
                }
            }
```

There are ways to write this more memory efficiently (string smashing is a very argued topic), but I just took a *similar* level of memory efficiency already there and main the 'main path' always encode `name`. Also, you only need to name the tuple elements if you actually reference them. Also, deconstruction should use lowercase names since you're effectively declaring local variables. This also could be done as Linq with a single statement, but I opted for local function as that is the preference for this repo.

Reply:
Or closer to what was there initially:

```suggestion
            return string.Join(",", pairs.Select(EncodePair));
            
            static string EncodePair((string Name, string Value) pair)
            {
                if (string.IsNullOrEmpty(Value))
                {
                    return EncodeCharacters(Name);
                }
                else
                {
                    return $"{EncodeCharacters(Name)}={EncodeCharacters(Value)}";
                }
            }
```